src/client/auth.ts

```diff
@@ -478,6 +497,178 @@ export function extractResourceMetadataUrl(res: Response): URL | undefined {
   }
 }
 
+/**
+ * Extracts required scopes from WWW-Authenticate header following SEP-835.
+ * 
+ * This function parses the WWW-Authenticate header to extract OAuth scopes
+ * that should be requested when authenticating. According to SEP-835, 
+ * this provides contextual guidance on the minimum scopes needed.
+ * 
+ * @param res - HTTP response containing WWW-Authenticate header
+ * @returns Array of required scope strings, or undefined if not present
+ */
+export function extractScopesFromWwwAuthenticate(res: Response): string[] | undefined {
+  const authenticateHeader = res.headers.get("WWW-Authenticate");
+  if (!authenticateHeader) {
+    return undefined;
+  }
+
+  const [type, scheme] = authenticateHeader.split(' ');
+  if (type.toLowerCase() !== 'bearer' || !scheme) {
+    return undefined;
+  }
+
+  // Parse scope parameter from Bearer scheme
+  // Format: Bearer realm="mcp", scope="scope1 scope2 scope3"
+  const scopeRegex = /scope="([^"]*)"/;
+  const match = scopeRegex.exec(authenticateHeader);
+
+  if (!match || !match[1]) {
+    return undefined;
+  }
+
+  // Split space-separated scopes and filter out empty strings
+  const scopes = match[1].split(' ').filter(scope => scope.trim().length > 0);
+  return scopes.length > 0 ? scopes : undefined;
```

Comment:
A token with value `Bearer realm="mcp", scope=""` is going to return `undefined` just like if scope parameter was not defined in the first place. Shouldn't we distinguish and return an empty array in this instance?
```suggestion
  return match[1].split(' ').filter(scope => scope.trim().length > 0);
```